core/node/da_clients/src/eigen/sdk.rs

```diff
@@ -244,3 +427,55 @@ fn convert_by_padding_empty_byte(data: &[u8]) -> Vec<u8> {
     valid_data.truncate(valid_end);
     valid_data
 }
+
+fn remove_empty_byte_from_padded_bytes(data: &[u8]) -> Vec<u8> {
+    let parse_size = DATA_CHUNK_SIZE;
+
+    // Calculate the number of chunks
+    let data_len = (data.len() + parse_size - 1) / parse_size;
```

Comment:
- Nit: There's [`div_ceil` family of methods](https://doc.rust-lang.org/core/primitive.usize.html#method.div_ceil) stabilized somewhat recently; you can use it here.
- Bikeshedding: The var name looks ambiguous (e.g., it could be interpreted as length in bytes); I'd rename it `chunk_count`.